### PR TITLE
Using honcho, adhering 12 factors

### PR DESCRIPTION
**Pull request by wazaundtechnik**
Locally we use Procfile too.

### CODE REVIEW

File: .envexample [hunk 1]
```diff
@@ -0,0 +1,3 @@
+TELEGRAM_BOT_TOKEN=yourtelegramtokenhere
+DEBUG=true
+LOG_LEVEL=DEBUG
```
Comment (kemskems) [inline .envexample]: `no newline at end of file`

File: README.md [hunk 2]
```diff
@@ -47,25 +47,16 @@
 
    Dependencies are all listed in `requirements.txt`. To generate this file, simply run `pip freeze > requirements.txt`.
 
-1. Before running the app, you have to set environment variables. Those variables will be used to configure the app. Again, there are many ways to do this one of them is to use virtualenv's `postactivate` and `predeactivate` hook.
+1. Before running the app, you have to set environment variables. Copy `.envexample` to `.env` file and make sure it suits your configuration
```
Comment (fushar) [inline README.md]: No full stop and the end of sentence.

**Comment by fushar** [comment]
Please create an issue describing why we should use Honcho or something like that.

**Comment by kemskems** [comment]
Ada satu concern fid. Kalau lagi nge-dev, enakan nge-run-nya itu `python app.py` supaya jalan di debug mode (gak pakai gunicorn). Salah satu keuntungannya adalah kalo ada uncaught exception, si Flask bisa tangkap dan kasih stack trace + REPL di browser (assuming via browser nembaknya). Kalau pakai gunicorn ini jadi gak bisa.

Salah satu solusinya adalah, karena config juga dah ditaruh di `.env`, di file `app.py` `.env`-nya di-load dulu. Ada library yg bisa dipake: https://pypi.python.org/pypi/python-dotenv. Tapi cara ini agak aneh juga sih kalo pake honcho jadi 2x ngebaca `.env`.

Ada ide lain?

**Comment by wazaundtechnik** [comment]
> Please create an issue describing why we should use Honcho or something like that.
So we need to create issue first? We can't straight into PR for improvement? Basically we want exact 

> Kalau lagi nge-dev, enakan nge-run-nya itu python app.py supaya jalan di debug mode (gak pakai gunicorn). Salah satu keuntungannya adalah kalo ada uncaught exception... (truncated)

I am aware with that. Can you tell me how to simulate exception? Actually, by default `DEBUG=true` for the `.env` file, but yes, still, it uses gunicorn.

File: README.md (hunk 2 above)
Comment (wazaundtechnik) [inline README.md]: Fixed.

File: .envexample (hunk 1 above)
Comment (wazaundtechnik) [inline .envexample]: Fixed

**Comment by kemskems** [comment]
@wazaundtechnik simulate exception? kasih aja `raise Exception()` di dalem view-nya.

**Comment by wazaundtechnik** [comment]
> @wazaundtechnik simulate exception? kasih aja raise Exception() di dalem view-nya.

Thanks kems. I will try that. I will make sure the REPL debugger still runs even if we run under Honcho.

**Comment by wazaundtechnik** [comment]
Hi all, please go to #15 to talk about the fundamental of using Local Procfile. Please make this PR clean by only discussing the code.

**Comment by wazaundtechnik** [comment]
This is obsolete by #39.

**Pull request closed without merging**